pkg/orchestration/wiring/datadog/plugin.go

```diff
@@ -0,0 +1,174 @@
+	instanceID, err := svccatentangler.InstanceID(resource.Spec)
+	if err != nil {
+		return smith_v1.Resource{}, errors.WithStack(err)
+	}
+	query := QueryParams{
+		KubeDeployment: kubeDeployment.Data.DeploymentName,
+		KubeNamespace:  context.StateMeta.Namespace,
+		AlarmType:      alarmType,
+		Threshold: &AlarmThresholds{
+			Critical: 90,
+			Warning:  80,
+		},
+		Location: context.StateContext.Location,
+	}
+	message := query.generateMessage(&context.StateContext.ServiceProperties.Notifications)
+	serviceInstanceSpec := OSBInstanceParameters{
+		ServiceName: context.StateContext.ServiceName,
+		Environment: context.StateContext.Location.EnvType,
+		Region:      context.StateContext.Location.Region,
+		Label:       context.StateContext.Location.Label,
+		Attributes: AlarmAttributes{
+			Name: createAlarmNameforDatadog(context.StateContext.ServiceName, context.StateContext.Location.EnvType,
+				resource.Name, query.AlarmType, context.StateContext.Location.Label),
+			Type:    Metric,
+			Query:   query.generateQuery(),
+			Message: message,
+			Options: AlarmOptions{
+				EscalationMessage: message,
+				NotifyNoData:      false,
+				RequireFullWindow: true,
+				Thresholds:        query.Threshold,
+			},
+		},
+	}
+
+	serviceInstanceSpecBytes, err := json.Marshal(&serviceInstanceSpec)
+	if err != nil {
+		return smith_v1.Resource{}, errors.WithStack(err)
+	}
+
+	alarmInstanceResource := smith_v1.Resource{
+		Name: wiringutil.ConsumerProducerResourceNameWithPostfix(resource.Name, kubeCompute, string(alarmType)),
+		References: []smith_v1.Reference{
+			{
+				Resource: kubeDeployment.Data.DeploymentResourceName,
+			},
+		},
+		Spec: smith_v1.ResourceSpec{
+			Object: &sc_v1b1.ServiceInstance{
+				TypeMeta: meta_v1.TypeMeta{
+					Kind:       "ServiceInstance",
+					APIVersion: sc_v1b1.SchemeGroupVersion.String(),
+				},
+				ObjectMeta: meta_v1.ObjectMeta{
+					Name: wiringutil.ConsumerProducerMetaNameWithPostfix(resource.Name, kubeCompute, string(alarmType)),
+				},
+				Spec: sc_v1b1.ServiceInstanceSpec{
+					PlanReference: sc_v1b1.PlanReference{
+						ClusterServiceClassExternalID: clusterServiceClassExternalID,
+						ClusterServicePlanExternalID:  clusterServicePlanExternalID,
+					},
+					Parameters: &runtime.RawExtension{
+						Raw: serviceInstanceSpecBytes,
+					},
+					ExternalID: instanceID,
+				},
+			},
+		},
+	}
+	return alarmInstanceResource, nil
+}
+
+func validateRequest(stateResource *orch_v1.StateResource, context *wiringplugin.WiringContext) error {
+	if stateResource.Type != ResourceType {
+		return errors.Errorf("invalid resource type: %q", stateResource.Type)
+	}
+	if stateResource.Spec != nil {
+		return errors.New("default alarm does not accept any user parameters")
+	}
+	return nil
+}
+
+func (q *QueryParams) generateQuery() string {
```

Comment:
This doesn't take label into consideration. I think it should.

Reply:
Won't the label be taken care of by virtue of being in the `Namespace`? Or is that strictly the Service Name, even if it's in a label?

Reply:
You are right, it's in the namespace. I wonder if instead of the namespace we should filter by service name + label.